addons/calendar/models/mail_activity_mixin.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+class MailActivityMixin(models.AbstractModel):
+    _inherit = 'mail.activity.mixin'
+
+    activity_calendar_event_id = fields.Many2one(
+        'calendar.event', "Next Activity Calendar Event", compute='_compute_activity_calendar_event_id',
```

Comment:
Linting: compute could go on the next line to keep line length under control.

I would also add string= before "Next Activity" (I like named parameters in field definition).